Tests/FoundationInternationalizationTests/Formatting/NumberFormatStyleTests.swift

```diff
@@ -1905,23 +1905,33 @@ final class FormatStylePatternMatchingTests : XCTestCase {
         _verifyMatching("€52,249", formatStyle: floatStyle, expectedUpperBound: nil, expectedValue: nil)
         _verifyMatching("€52,249", formatStyle: decimalStyle, expectedUpperBound: nil, expectedValue: nil)
 
-        // Different locale
         let frenchStyle: IntegerFormatStyle<Int>.Currency = .init(code: "EUR", locale: frFR)
-
-        let frenchPrice = "57 379 €"
-        _verifyMatching(frenchPrice, formatStyle: frenchStyle, expectedUpperBound: frenchPrice.endIndex, expectedValue: 57379)
-        _verifyMatching(frenchPrice, formatStyle: floatStyle.locale(frFR), expectedUpperBound: frenchPrice.endIndex, expectedValue: 57379)
```

Comment:
The tests incorrectly assumed that a currency style with "USD" code but "fr_FR" locale can correctly match a currency string with EUR symbol (`"57 379,00 €"`). This isn't correct, and is exactly what this bug is about: we only respect the locale regional currency, but ignore the currency code.